service/controller/resource/tcnp/template/template_main_security_groups.go

```diff
@@ -81,5 +81,17 @@ const TemplateMainSecurityGroups = `
       FromPort: -1
       ToPort: -1
       SourceSecurityGroupId: {{ .SecurityGroups.TenantCluster.Master.ID }}
+  {{ range .SecurityGroups.TenantCluster.NodePools }}
+  NodePoolToNodePoolRule{{ .ResourceName }}:
+    Type: AWS::EC2::SecurityGroupIngress
+    DependsOn: GeneralSecurityGroup
+    Properties:
+      Description: NodePoolToNodePool - Allow traffic from the other NodePool Security Group to the this Node Pool General Security Group.
```

Comment:
```suggestion
      Description: Allow traffic from other Node Pool Security Groups to this Node Pool's Security Group.
```

Reply:
the codeword NodePoolToNodePool is important as its used to filter the right security group later